### PR TITLE
[CF-87] Show global progress of Cinder Volume migration

### PR DESCRIPTION
**Pull request by vladiskuz**
Add some output information in logs about global migration
process of Cinder volumes.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/2280/

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/os/actions/cinder_database_manipulation.py [hunk 1]
```diff
@@ -109,6 +111,75 @@ def _clean_data(data):
     return data
 
 
+class CinderStorageMigrationProgressView(object):
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/cinder_database_manipulation.py]: Minor: might be better to move this class to a separate file, migration progress does not seem to be a part of  `cinder_database_manipulation` module

File: cloudferrylib/os/actions/cinder_database_manipulation.py [hunk 2]
```diff
@@ -109,6 +111,75 @@ def _clean_data(data):
...
+        total_volumes_size_gb = self.total_volumes_size_b/(1024.0**3)
+        migrated_volumes_size_gb = self.migrated_volumes_size_b/(1024.0**3)
+        failed_volumes_size_gb = self.failed_volumes_size_b/(1024.0**3)
+
+        LOG.info('Number of migrated images %d of %d. '
+                 'Volumes migrated %.1f%% and '
+                 'failed %.1f%% of %.1f GB total at %.1f MB/s.',
+                 self.num_migrated,
+                 len(self.src_volumes) + len(self.dst_volumes),
+                 migrated_volumes_size_gb/total_volumes_size_gb*100,
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/cinder_database_manipulation.py]: Would be much better to calculate speeds dynamically as we did for image copy: `cloudferrylib/utils/sizeof_format.py`

File: cloudferrylib/os/actions/cinder_database_manipulation.py [hunk 3]
```diff
@@ -109,6 +111,75 @@ def _clean_data(data):
...
+    def sync_failed_volumes_info(self, volume):
+        self.failed_volumes_size_b += self.volumes_size_map[volume['id']]
+        self.failed_volumes.append(volume)
+
+    def show_progress(self):
+        total_volumes_size_gb = self.total_volumes_size_b/(1024.0**3)
+        migrated_volumes_size_gb = self.migrated_volumes_size_b/(1024.0**3)
+        failed_volumes_size_gb = self.failed_volumes_size_b/(1024.0**3)
+
+        LOG.info('Number of migrated images %d of %d. '
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/cinder_database_manipulation.py]: Volumes, not images

File: cloudferrylib/os/actions/cinder_database_manipulation.py (hunk 1 above)
Comment (vladiskuz) [inline cloudferrylib/os/actions/cinder_database_manipulation.py]: Done

File: cloudferrylib/os/actions/cinder_database_manipulation.py (hunk 2 above)
Comment (vladiskuz) [inline cloudferrylib/os/actions/cinder_database_manipulation.py]: Done

File: cloudferrylib/os/actions/cinder_database_manipulation.py (hunk 3 above)
Comment (vladiskuz) [inline cloudferrylib/os/actions/cinder_database_manipulation.py]: Done

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/2291/

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
+1

**Comment by archyufa** [comment]
Is it possible to make INFO message of progress for each volume transfer from share to share ? We generally want to have INFO mode intaractive. But right now when we have rsync of volume it looks like the tool is hanging and doing nothing in INFO mode.

**Comment by vladiskuz** [comment]
@archyufa with this pull request if we run CF without debug mode 

> fab migrate:configuration.ini (without debug=True)

 we can see the following in the console output:
```
INFO     WriteVolumesDb [/home/vk/projects/CloudFerry/cloudferrylib/os/actions/cinder_database_manipulation.py:463] 01-28 12:54:54 Enough space found on /var/lib/cinder/dc00daa166ae102ec46ace7fad12ba05
[192.168.1.2] run: rsync --progress --partial --inplace --perms --times -e "ssh -o StrictHostKeyChecking=no" /var/lib/cinder/96af0807784460ad580b59cb9a75bc93/volume-7467f42c-799c-454c-801b-2cc780117a22 root@192.168.1.3:/var/lib/cinder/dc00daa166ae102ec46ace7fad12ba05
[192.168.1.2] out: volume-7467f42c-799c-454c-801b-2cc780117a22
[192.168.1.2] out: 
[192.168.1.2] out:        32768   0%    0.00kB/s    0:00:00  
[192.168.1.2] out:     11370496   1%   10.33MB/s    0:01:40 
[192.168.1.2] out:     25591808   2%   11.91MB/s    0:01:25  
[192.168.1.2] out:     39550976   3%   12.29MB/s    0:01:22  
[192.168.1.2] out:     50724864   4%   11.83MB/s    0:01:24  
[192.168.1.2] out:     67764224   6%   13.29MB/s    0:01:13  
...
```
The output from rsync with percentage, speed and time estimation write in stdout fabric. We don't have this part in an output log file. In the log output file we have the following information:
```
INFO     WriteVolumesDb 01-28 12:42:11 Total number of volumes to be migrated: 7
INFO     WriteVolumesDb 01-28 12:42:11 List of volumes to be migrated:
tn3_volume1(5aed7997-563a-4706-ad79-0ade690a408e)
tn1_volume2(5b9c035e-9e8b-4780-a5d9-a17c77969d78)
tn1_volume1(7467f42c-799c-454c-801b-2cc780117a22)
cinder_volume2(8821bc29-fb46-4934-bb51-6c944f329155)
cinder_volume3(cc55dae0-0c2c-4a56-8f41-023c94a08237)
cinder_volume1(d571e3b7-bfed-4e04-a9a0-e24c7ab92aef)
tn_volume1(f615df6d-ca31-4f64-aa39-6421a73807d0)
INFO     WriteVolumesDb 01-28 12:42:11 Migrating volume: tn3_volume1(5aed7997-563a-4706-ad79-0ade690a408e)
INFO     WriteVolumesDb 01-28 12:42:18 Copying volume file: /var/lib/cinder/5326fb5519951f28556401e184d5b9d1/volume-5aed7997-563a-4706-ad79-0ade690a408e
INFO     WriteVolumesDb 01-28 12:42:42 Enough space found on /var/lib/cinder/dc00daa166ae102ec46ace7fad12ba05
INFO     WriteVolumesDb 01-28 12:47:04 Number of migrated volumes 1 of 7. Volumes migrated 14.3% and failed 0.0% of 7.0GB total at 3.7 MB/s.
INFO     WriteVolumesDb 01-28 12:47:04 Migrating volume: tn1_volume2(5b9c035e-9e8b-4780-a5d9-a17c77969d78)
INFO     WriteVolumesDb 01-28 12:47:09 Copying volume file: /var/lib/cinder/5326fb5519951f28556401e184d5b9d1/volume-5b9c035e-9e8b-4780-a5d9-a17c77969d78
INFO     WriteVolumesDb 01-28 12:47:11 Enough space found on /var/lib/cinder/42efebc0e06c19d8773b47a06b8a18bd
INFO     WriteVolumesDb 01-28 12:48:24 Number of migrated volumes 2 of 7. Volumes migrated 28.6% and failed 0.0% of 7.0GB total at 13.9 MB/s.
```
From the log file we can see status in global. I think that percentage, speed and time estimation are not so important information, that should be written in the log file. I don't want to overload the info log file. What do you think about that?

**Comment by archyufa** [comment]
As we disccused can you update info log part to make it readable and make some sense, as well as it has some logical start and end. The rest looks good!

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/2412/

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by vladiskuz** [comment]
@archyufa 
I've fixed the log output, right now it looks like output below:
```
INFO     WriteVolumesDb 01-29 14:35:31 Start task 'WriteVolumesDb'
INFO     WriteVolumesDb 01-29 14:35:31 Start volumes migration process.
INFO     WriteVolumesDb 01-29 14:35:31 Filtered volumes: tn1_volume1, cinder_volume2, cinder_volume_error, cinder_volume3, tn_volume1, cinder_volume1, cinder_volume_creating, tn1_volume2, cinder_volume_err
or_deleting, cinder_volume_deleting, tn3_volume1
WARNING  WriteVolumesDb 01-29 14:35:31 Skipping volume cinder_volume_error[724f1d3f-aa20-450e-a441-5df3f3ce00cf] in "error" state
WARNING  WriteVolumesDb 01-29 14:35:31 Skipping volume cinder_volume_creating[dfe73eb3-db46-4b1d-99a8-9e6b0bcf60eb] in "creating" state
WARNING  WriteVolumesDb 01-29 14:35:31 Skipping volume cinder_volume_error_deleting[ee5d807e-0005-4e30-a8ff-23ab53525175] in "error_deleting" state
WARNING  WriteVolumesDb 01-29 14:35:31 Skipping volume cinder_volume_deleting[ef73e94a-c0b1-4487-8d6e-edeae381fb14] in "deleting" state
INFO     WriteVolumesDb 01-29 14:35:31 User's 'user6' primary tenant '6c1d690f84c347588693f8ee9d8234a6' is deleted, finding out if user is a member of other tenants
INFO     WriteVolumesDb 01-29 14:35:31 Setting tenant 'tenant1' for user 'user6' as primary
INFO     WriteVolumesDb 01-29 14:35:31 User's 'user5' primary tenant '6c1d690f84c347588693f8ee9d8234a6' is deleted, finding out if user is a member of other tenants
INFO     WriteVolumesDb 01-29 14:35:31 User's 'user3' primary tenant 'None' is deleted, finding out if user is a member of other tenants
INFO     WriteVolumesDb 01-29 14:35:31 User's 'user8' primary tenant '6c1d690f84c347588693f8ee9d8234a6' is deleted, finding out if user is a member of other tenants
INFO     WriteVolumesDb 01-29 14:35:34 User's 'user6' primary tenant '6c1d690f84c347588693f8ee9d8234a6' is deleted, finding out if user is a member of other tenants
INFO     WriteVolumesDb 01-29 14:35:34 Setting tenant 'tenant1' for user 'user6' as primary
INFO     WriteVolumesDb 01-29 14:35:34 User's 'user5' primary tenant '6c1d690f84c347588693f8ee9d8234a6' is deleted, finding out if user is a member of other tenants
INFO     WriteVolumesDb 01-29 14:35:34 User's 'user3' primary tenant 'None' is deleted, finding out if user is a member of other tenants
INFO     WriteVolumesDb 01-29 14:35:34 User's 'user8' primary tenant '6c1d690f84c347588693f8ee9d8234a6' is deleted, finding out if user is a member of other tenants
INFO     WriteVolumesDb 01-29 14:35:35 Filtered quotas: 
INFO     WriteVolumesDb 01-29 14:35:35 Filtered quota_classes: 
INFO     WriteVolumesDb 01-29 14:35:35 Start compare existing volumes on the destination cloud. If volumes exist on the destination cloud, then skip migration for those volumes.
INFO     WriteVolumesDb 01-29 14:35:35 Volume tn1_volume1(0ffacf1e-f578-4321-aced-efd621e83c23) does not exist on the destination cloud, will be migrated.
INFO     WriteVolumesDb 01-29 14:35:35 Volume cinder_volume2(369ea4c5-649d-4d87-9f9b-15170271b1d6) does not exist on the destination cloud, will be migrated.
INFO     WriteVolumesDb 01-29 14:35:35 Volume cinder_volume3(7899c821-273d-47be-9c40-4d5dfaba8050) does not exist on the destination cloud, will be migrated.
INFO     WriteVolumesDb 01-29 14:35:35 Volume tn_volume1(8f5e0c34-ca53-42c3-8be8-6c7e10c4548f) does not exist on the destination cloud, will be migrated.
INFO     WriteVolumesDb 01-29 14:35:35 Volume cinder_volume1(b3391ec5-5f12-4c79-aaee-f68609725d70) does not exist on the destination cloud, will be migrated.
INFO     WriteVolumesDb 01-29 14:35:35 Volume tn1_volume2(e9919312-78bc-4df6-9e06-81e1b3b6606c) does not exist on the destination cloud, will be migrated.
INFO     WriteVolumesDb 01-29 14:35:35 Volume tn3_volume1(f71427f5-874c-4af0-94b9-15ab01b6cefe) does not exist on the destination cloud, will be migrated.
INFO     WriteVolumesDb 01-29 14:35:35 All volumes on the source and destination cloud have been compared.
INFO     WriteVolumesDb 01-29 14:35:35 Calculate size of each volume.
INFO     WriteVolumesDb 01-29 14:35:37 Volume tn1_volume1(0ffacf1e-f578-4321-aced-efd621e83c23) size is 1.0GB.
INFO     WriteVolumesDb 01-29 14:35:37 Volume cinder_volume2(369ea4c5-649d-4d87-9f9b-15170271b1d6) size is 1.0GB.
INFO     WriteVolumesDb 01-29 14:35:37 Volume cinder_volume3(7899c821-273d-47be-9c40-4d5dfaba8050) size is 1.0GB.
INFO     WriteVolumesDb 01-29 14:35:37 Volume tn_volume1(8f5e0c34-ca53-42c3-8be8-6c7e10c4548f) size is 1.0GB.
INFO     WriteVolumesDb 01-29 14:35:38 Volume cinder_volume1(b3391ec5-5f12-4c79-aaee-f68609725d70) size is 1.0GB.
INFO     WriteVolumesDb 01-29 14:35:38 Volume tn1_volume2(e9919312-78bc-4df6-9e06-81e1b3b6606c) size is 1.0GB.
INFO     WriteVolumesDb 01-29 14:35:38 Volume tn3_volume1(f71427f5-874c-4af0-94b9-15ab01b6cefe) size is 1.0GB.
INFO     WriteVolumesDb 01-29 14:35:38 Total number of volumes to be migrated: 7
INFO     WriteVolumesDb 01-29 14:35:38 List of volumes to be migrated:
tn1_volume1(0ffacf1e-f578-4321-aced-efd621e83c23)
cinder_volume2(369ea4c5-649d-4d87-9f9b-15170271b1d6)
cinder_volume3(7899c821-273d-47be-9c40-4d5dfaba8050)
tn_volume1(8f5e0c34-ca53-42c3-8be8-6c7e10c4548f)
cinder_volume1(b3391ec5-5f12-4c79-aaee-f68609725d70)
tn1_volume2(e9919312-78bc-4df6-9e06-81e1b3b6606c)
tn3_volume1(f71427f5-874c-4af0-94b9-15ab01b6cefe)
INFO     WriteVolumesDb 01-29 14:35:38 Start migrate volume tn1_volume1(0ffacf1e-f578-4321-aced-efd621e83c23)
INFO     WriteVolumesDb 01-29 14:35:38 Calculate free space on the destination cloud.
INFO     WriteVolumesDb 01-29 14:35:38 Enough space found on /var/lib/cinder/dc00daa166ae102ec46ace7fad12ba05
INFO     WriteVolumesDb 01-29 14:35:38 Start copying volume.
INFO     WriteVolumesDb 01-29 14:36:51 Volume tn1_volume1(0ffacf1e-f578-4321-aced-efd621e83c23) has been successfully migrated.
INFO     WriteVolumesDb 01-29 14:36:51 Number of migrated volumes 1 of 7. Volumes migrated 14.3% and failed 0.0% of 7.0GB total at 14.0 MB/s.
INFO     WriteVolumesDb 01-29 14:36:51 Start migrate volume cinder_volume2(369ea4c5-649d-4d87-9f9b-15170271b1d6)
INFO     WriteVolumesDb 01-29 14:36:51 Calculate free space on the destination cloud.
INFO     WriteVolumesDb 01-29 14:36:51 Enough space found on /var/lib/cinder/42efebc0e06c19d8773b47a06b8a18bd
INFO     WriteVolumesDb 01-29 14:36:51 Start copying volume.
INFO     WriteVolumesDb 01-29 14:38:03 Volume cinder_volume2(369ea4c5-649d-4d87-9f9b-15170271b1d6) has been successfully migrated.
INFO     WriteVolumesDb 01-29 14:38:03 Number of migrated volumes 2 of 7. Volumes migrated 28.6% and failed 0.0% of 7.0GB total at 14.4 MB/s.
INFO     WriteVolumesDb 01-29 14:38:03 Start migrate volume cinder_volume3(7899c821-273d-47be-9c40-4d5dfaba8050)
INFO     WriteVolumesDb 01-29 14:38:03 Calculate free space on the destination cloud.
INFO     WriteVolumesDb 01-29 14:38:03 Enough space found on /var/lib/cinder/dc00daa166ae102ec46ace7fad12ba05
INFO     WriteVolumesDb 01-29 14:38:03 Start copying volume.
```

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by archyufa** [comment]
+1 Can be merged

**Comment by roman-verchikov** [comment]
Please squash two commits into one

**Comment by MirantisWorkloadMobility** [comment]
$PULL_STATUS
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/2450/

**Comment by MirantisWorkloadMobility** [comment]
$PYLINT_STATUS

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
can be merged

**Pull request merged**